lms/djangoapps/courseware/masquerade.py

```diff
@@ -71,27 +72,31 @@ def handle_ajax(request, course_key_string):
     group_id = request_json.get('group_id', None)
     user_partition_id = request_json.get('user_partition_id', None) if group_id is not None else None
     user_name = request_json.get('user_name', None)
+    # The 'user_name' param is actually a string representing either the username or the user's
+    # email address. 'found_user_name' represents the actual username property for a User.
+    found_user_name = None
+    # This view should return an error if no 'user_name' was provided, but many different test cases
+    # rely on the view still working if 'user_name' is blank or None.
     if user_name:
         users_in_course = CourseEnrollment.objects.users_enrolled_in(course_key)
         try:
-            if '@' in user_name:
-                user_name = users_in_course.get(email=user_name).username
-            else:
-                users_in_course.get(username=user_name)
+            # A username can have an '@' symbol in it, so if no user was found with an email that
```

Comment:
Please move the comment below to tests:

> A username can have an '@' symbol in it, so if no user was found with an email that
> matches the 'user_name' string, we still want to check if it matches a username property.

It's clearer there (check the tests file for more details).